Wikipedia/UI-V5/WMFRelatedTitleListDataSource.h

```diff
@@ -0,0 +1,37 @@
+//
+//  WMFRelatedTitleListDataSource.h
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/4/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <SSDataSources/SSArrayDataSource.h>
+#import "WMFArticleListDataSource.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class WMFRelatedSearchFetcher;
+@class MWKSavedPageList;
+@class MWKDataStore;
+@class MWKTitle;
+
+@interface WMFRelatedTitleListDataSource : SSArrayDataSource
+    <WMFArticleListDataSource>
+
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters;
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters
+                      fetcher:(WMFRelatedSearchFetcher*)fetcher NS_DESIGNATED_INITIALIZER;
```

Comment:
Here it isn't about danger… its about exposing the smallest public API required. Which is one of our best practices.

https://www.mediawiki.org/wiki/Wikimedia_Apps/Team/iOS/BestPractices#What_should_be_public

This also goes back to what testing is for… which is to support development… methods for testing are just that… and should not be exposed to all users of a class.